browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/speedreader/speedreader_tab_helper.h"
+
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/browser/speedreader/speedreader_service_factory.h"
+#include "brave/components/speedreader/speedreader_service.h"
+#include "brave/components/speedreader/speedreader_test_whitelist.h"
+#include "brave/components/speedreader/speedreader_whitelist.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/public/browser/navigation_handle.h"
+
+namespace speedreader {
+
+SpeedreaderTabHelper::~SpeedreaderTabHelper() = default;
+
+SpeedreaderTabHelper::SpeedreaderTabHelper(content::WebContents* web_contents)
+    : content::WebContentsObserver(web_contents) {}
+
+void SpeedreaderTabHelper::UpdateActiveState(
+    content::NavigationHandle* handle) {
+  Profile* profile =
+      Profile::FromBrowserContext(web_contents()->GetBrowserContext());
+  DCHECK(profile);
+  const bool enabled =
+      SpeedreaderServiceFactory::GetForProfile(profile)->IsEnabled();
+
+  if (!enabled) {
+    active_ = false;
```

Comment:
maybe we should add `return false;` here?
I assume active state is always false when preference is off. isn't it?

Reply:
absolutely

Reply:
actually it was caught by the test :D